src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/AutoInsert/AutoClosingTagOnAutoInsertProvider.cs

```diff
@@ -38,70 +40,51 @@ internal sealed class AutoClosingTagOnAutoInsertProvider : IOnAutoInsertProvider
     );
     private static readonly ImmutableHashSet<string> s_voidElementsCaseSensitive = s_voidElements.WithComparer(StringComparer.Ordinal);
 
-    private readonly RazorLSPOptionsMonitor _optionsMonitor;
     private readonly ILogger _logger;
 
-    public AutoClosingTagOnAutoInsertProvider(RazorLSPOptionsMonitor optionsMonitor, ILoggerFactory loggerFactory)
+    public AutoClosingTagOnAutoInsertProvider(ILoggerFactory loggerFactory)
     {
-        if (optionsMonitor is null)
-        {
-            throw new ArgumentNullException(nameof(optionsMonitor));
-        }
-
         if (loggerFactory is null)
         {
             throw new ArgumentNullException(nameof(loggerFactory));
         }
 
-        _optionsMonitor = optionsMonitor;
         _logger = loggerFactory.GetOrCreateLogger<IOnAutoInsertProvider>();
     }
 
     public string TriggerCharacter => ">";
 
-    public bool TryResolveInsertion(Position position, FormattingContext context, [NotNullWhen(true)] out TextEdit? edit, out InsertTextFormat format)
+    public async ValueTask<InsertTextEdit?> TryResolveInsertionAsync(Position position, IDocumentSnapshot documentSnapshot, bool autoClosingTagsOption)
```

Comment:
Nit: I know the options property it came from is named terribly, but the last parameter here would be much nicer if it was `enableAutoClosingTags` or something :)

Reply:
Yeah, I think I dropped "option" elsewhere, agree it sounded awkward. Will use the suggested name.